compiler/hash-typecheck/src/ops/validate.rs

```diff
@@ -55,22 +55,198 @@ enum MergeKind {
     Level1 { nominal_attached: Option<TermId> },
 }
 
+/// Helper type for [Validator::validate_constant_scope], to determine in what
+/// capacity the `Self` member should be existing in the scope.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[allow(unused)] // @@Todo: remove
+enum SelfMode {
+    Allowed,
+    Required,
+}
+
 impl<'gs, 'ls, 'cd> Validator<'gs, 'ls, 'cd> {
     pub fn new(storage: StorageRefMut<'gs, 'ls, 'cd>) -> Self {
         Self { storage }
     }
 
-    /// Validate the module definition of the given [ModDefId]
-    pub fn validate_mod_def(&mut self, _mod_def_id: ModDefId) -> TcResult<()> {
+    /// Validate the members of the given constant scope.
+    ///
+    /// Allows uninitialised members in the scope if `allow_uninitialised` is
+    /// true.
+    fn validate_constant_scope(
+        &mut self,
+        scope_id: ScopeId,
+        allow_uninitialised: bool,
+    ) -> TcResult<()> {
+        // @@Design: when do we insert each member into the scope? As we go or all at
+        // once? For now, we insert as we go.
+
+        // Enter the progressive scope:
+        let progressive_scope = Scope::new(ScopeKind::Constant, []);
+        let progressive_scope_id = self.scope_store_mut().create(progressive_scope);
+        self.scopes_mut().append(progressive_scope_id);
+
+        // @@Performance: sad that we have to clone here:
+        let scope = self.reader().get_scope(scope_id).clone();
+        for member in scope.iter() {
+            // This should have been checked in semantic analysis:
+            assert!(
+                member.mutability == Mutability::Mutable,
```

Comment:
```suggestion
                member.mutability != Mutability::Mutable,
```
 Checking it's immutable

Reply:
OOF